File: week1-cpp/Application_Sunglasses_Filter/Using_Naive_Replace.cpp
#include <iostream>
#include <opencv2/core.hpp>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/opencv.hpp>
#include "dataPath.hpp"

using namespace std;
using namespace cv;

int main(void){
        // Load the Face Image
        string faceImagePath = DATA_PATH + "/images/musk.jpg";

        Mat faceImage = imread(faceImagePath);

	Mat glassBGR = imread("../results/sunglassRGB.png");
	// Make a copy
	Mat faceWithGlassesNaive = faceImage.clone();
	Mat roiFace =  faceWithGlassesNaive(Range(150,250),Range(140,440));
	// Replace the eye region with the sunglass image
	glassBGR.copyTo(roiFace);

	imwrite("../results/faceWithGlassesNaive.png",faceWithGlassesNaive);
	//imshow("Face with Glasses Naive",faceWithGlassesNaive);
	//waitKey(0);

	return 0;
}

File: week1-python/Application_Sunglasses_Filter/Using_Naive_Replace.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH

# Load the Face Image
faceImagePath = DATA_PATH + "/images/musk.jpg"
faceImage = cv2.imread(faceImagePath)

# Load the Sunglass image with Alpha channel
# (http://pluspng.com/sunglass-png-1104.html)
glassimagePath = DATA_PATH + "/images/sunglass.png"
glassPNG = cv2.imread(glassimagePath,-1)

# Resize the image to fit over the eye region
glassPNG = cv2.resize(glassPNG,(300,100))
print("image Dimension ={}".format(glassPNG.shape))

# Separate the Color and alpha channels
glassBGR = glassPNG[:,:,0:3]
glassMask1 = glassPNG[:,:,3]

# Make a copy
faceWithGlassesNaive = faceImage.copy()

# Replace the eye region with the sunglass image
faceWithGlassesNaive[150:250,140:440]=glassBGR

#cv2.imshow("naive replace",faceWithGlassesNaive)
#cv2.waitKey(0)
#cv2.destroyAllWindows()
cv2.imwrite("../results/faceWithGlassesNaive.png",faceWithGlassesNaive)

